teachertool/src/components/makecodeFrame.tsx

```diff
@@ -0,0 +1,138 @@
+/// <reference path="../../../built/pxteditor.d.ts" />
+import * as React from "react";
+import { useState, useEffect, useRef } from "react";
+import { ProgressBar } from "react-common/components/controls/ProgressBar";
+interface MakeCodeFrameProps {
+    pageSourceUrl: string;
+    highContrast?: boolean;
+    onFrameOpen: () => void;
+    onFrameClose: () => void;
+    tutorialEventHandler?: (event: pxt.editor.EditorMessageTutorialEventRequest) => void;
```

Comment:
What is tutorialEventHandler for?

Reply:
This is something that I want to have here for making the frame more generic.  The generic MakeCodeFrame is the thing that needs to listen for the tutorial event messages from the editor, but the way that tutorial messages are processed I think are going to vary. For example, skillmap handles tutorial events in ways that are very specific to skillmaps. It would be great if the generic MakeCodeFrame replaced, or could replace, all of our uses of the iframe, so this function would be needed as a prop if and when the frame would be used by skillmap so the function that processes tutorial events on skillmap's side can do so. 

The function that I'm referencing from skillmap is here: https://github.com/microsoft/pxt/blob/master/skillmap/src/components/makecodeFrame.tsx#L282-L309

You can see that each of the individual events in this function call functions that are specific for skillmap.